types/yt-search/index.d.ts

```diff
@@ -93,6 +93,20 @@ declare namespace yts {
         author: Pick<Author, 'name' | 'url'>;
     }
 
+    interface LiveSearchResult {
+        type: 'live';
+        title: string;
+        description: string;
+        url: string;
+        videoId: string;
+        watching: number;
+        image: string;
+        /** @deprecated */
+        thumbnail: string;
+        author: Pick<Author, 'name' | 'url'>;
+        status: 'LIVE' | 'UPCOMING';
+    }
```

Comment:
There's also the `startTime` and `startDate` properties if the livestream is upcoming: see https://github.com/talmobi/yt-search/blob/e782bc4a07e9eac35d8ad92f4d08af7a7e612c46/src/index.js#L734-L740

```suggestion
    interface LiveSearchResultBase {
        type: 'live';
        videoId: string;
        url: string;
        title: string;
        description: string;
        image: string;
        thumbnail: string;
        watching: number;
        author: Author;
    }

    interface UpcomingLiveSearchResult extends LiveSearchResultBase {
        status: 'UPCOMING';
        startTime: number
        startDate: string
    }

    interface LiveLiveSearchResult extends LiveSearchResultBase {
        status: 'LIVE';
    }

    type LiveSearchResult = UpcomingLiveSearchResult | LiveLiveSearchResult
```

There has also been a number of other changes since v2.0:

```diff
     interface Author {
         name: string;
-        id: string;
         url: string;
-        userId: string;
-        userUrl: string;
-        userName: string;
-        channelId: string;
-        channelUrl: string;
-        channelName: string;
    }
```

```diff
-    interface PlaylistAuthor extends Omit<Author, 'userName' | 'channelName'> {
-        userText: string;
-        channelText: string;
-    }
```

```diff
     interface VideoSearchResult {
         type: 'video';
         title: string;
         description: string;
         url: string;
         videoId: string;
         seconds: number;
         timestamp: string;
         duration: Duration;
         views: number;
         image: string;
-        /** @deprecated */
         thumbnail: string;
         ago: string;
-        author: Pick<Author, 'name' | 'url'>;
+        author: Author;
     }
```

```diff
     interface PlaylistSearchResult {
         type: 'list';
         title: string;
         url: string;
         listId: string;
-        videoCountLabel: string;
         videoCount: number;
         image: string;
-        /** @deprecated */
         thumbnail: string;
         author: Author;
     }
```

```diff
     interface ChannelSearchResult {
         type: 'channel';
         title: string;
-        description: string;
         url: string;
         videoCountLabel: string;
         videoCount: number;
         image: string;
-        /** @deprecated */
         thumbnail: string;
         name: string;
-        id: string;
+        subCount: number;
+        subCountLabel: string;
     }
```

```diff
     interface VideoMetadataResult {
         title: string;
         description: string;
         url: string;
         videoId: string;
         seconds: number;
         timestamp: string;
         duration: Duration;
         views: number;
         genre: string;
         uploadDate: string;
         ago: string;
         image: string;
-        /** @deprecated */
         thumbnail: string;
         author: Author;
     }
```

```diff
     interface PlaylistItem {
         title: string;
         videoId: string;
         listId: string;
-        url: string;
-        thumbnailUrl: string;
-        thumbnailUrlHQ: string;
-        owner: string;
-        author: PlaylistAuthor;
+        thumbnail: string;
+        author: Author;
     }
```
 
```diff
     interface PlaylistMetadataResult {
         title: string;
         listId: string;
         url: string;
-        /** @deprecated */
-        videoCount: number;
         views: number;
         date: string;
         image: string;
-        /** @deprecated */
         thumbnail: string;
-        items: PlaylistItem[];
         videos: PlaylistItem[];
-        author: PlaylistAuthor;
+        author: Author;
     }
 }
```

The `// DEPRECATED` comments for `thumbnail` were removed in the source code in [this commit](https://github.com/talmobi/yt-search/commit/962e57e5198edd75f335ee4bdf9437660b860ea9).

Additionally, because `Omit` isn't used anymore, the `// Minimum TypeScript Version: 3.5` in the header could be removed.

If you want, you can just use the following code. I also reordered the properties to match the source code for easier maintenance and changed `toString(): string` to `toString: () => string` in `Duration` because technically the method isn't bound ([1](https://github.com/talmobi/yt-search/blob/e782bc4a07e9eac35d8ad92f4d08af7a7e612c46/src/index.js#L1131), [2](https://github.com/talmobi/yt-search/blob/e782bc4a07e9eac35d8ad92f4d08af7a7e612c46/src/index.js#L1158), [3](https://github.com/talmobi/yt-search/blob/e782bc4a07e9eac35d8ad92f4d08af7a7e612c46/src/index.js#L1176), [4](https://github.com/talmobi/yt-search/blob/e782bc4a07e9eac35d8ad92f4d08af7a7e612c46/src/index.js#L1201)).

<details>

```ts
// Type definitions for yt-search 2.3
// Project: https://github.com/talmobi/yt-search#readme
// Definitions by: cherryblossom <https://github.com/cherryblossom000>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped

export = yts;

declare const yts: typeof search & { search: typeof search };

declare function search(query: string | yts.Options, callback: (err: Error | string | null | undefined, data: yts.SearchResult) => void): void;
declare function search(query: string | yts.Options): Promise<yts.SearchResult>;

declare function search(query: yts.VideoMetadataOptions, callback: (err: Error | string | null | undefined, data: yts.VideoMetadataResult) => void): void;
declare function search(query: yts.VideoMetadataOptions): Promise<yts.VideoMetadataResult>;

declare function search(query: yts.PlaylistMetadataOptions, callback: (err: Error | string | null | undefined, data: yts.PlaylistMetadataResult) => void): void;
declare function search(query: yts.PlaylistMetadataOptions): Promise<yts.PlaylistMetadataResult>;

declare namespace yts {
    interface BaseOptions {
        pageStart?: number;
        pageEnd?: number;
        pages?: number;
        userAgent?: string;

        /**
         * The language.
         * @default 'en'
         */
        hl?: string;

        /**
         * The location.
         * @default 'US'
         */
        gl?: string;

        /**
         * The category (for example `'music'`.)
         * @default ''
         */
        category?: string;

        sp?: string;
    }

    interface OptionsWithQuery extends BaseOptions { query: string; }
    interface OptionsWithSearch extends BaseOptions { search: string; }

    type Options = OptionsWithQuery | OptionsWithSearch;

    interface VideoMetadataOptions { videoId: string; }
    interface PlaylistMetadataOptions { listId: string; }

    interface Author {
        name: string;
        url: string;
    }

    interface Duration {
        seconds: number;
        timestamp: string;
        toString: () => string;
    }

    interface VideoSearchResult {
        type: 'video';
        videoId: string;
        url: string;
        title: string;
        description: string;
        image: string;
        thumbnail: string;
        seconds: number;
        timestamp: string;
        duration: Duration;
        ago: string;
        views: number;
        author: Author;
    }

    interface LiveSearchResultBase {
        type: 'live';
        videoId: string;
        url: string;
        title: string;
        description: string;
        image: string;
        thumbnail: string;
        watching: number;
        author: Author;
    }

    interface UpcomingLiveSearchResult extends LiveSearchResultBase {
        status: 'UPCOMING';
        startTime: number
        startDate: string
    }

    interface LiveLiveSearchResult extends LiveSearchResultBase {
        status: 'LIVE';
    }

    type LiveSearchResult = UpcomingLiveSearchResult | LiveLiveSearchResult

    interface PlaylistSearchResult {
        type: 'list';
        listId: string;
        url: string;
        title: string;
        image: string;
        thumbnail: string;
        videoCount: number;
        author: Author;
    }

    interface ChannelSearchResult {
        type: 'channel';
        name: string;
        url: string;
        title: string;
        image: string;
        thumbnail: string;
        videoCount: number;
        videoCountLabel: string;
        subCount: number;
        subCountLabel: string;
    }

    interface SearchResult {
        all: Array<VideoSearchResult | LiveSearchResult | PlaylistSearchResult | ChannelSearchResult>;
        videos: VideoSearchResult[];
        live: LiveSearchResult[];
        playlists: PlaylistSearchResult[];
        lists: PlaylistSearchResult[];
        accounts: ChannelSearchResult[];
        channels: ChannelSearchResult[];
    }

    interface VideoMetadataResult {
        title: string;
        description: string;
        url: string;
        videoId: string;
        seconds: number;
        timestamp: string;
        duration: Duration;
        views: number;
        genre: string;
        uploadDate: string;
        ago: string;
        image: string;
        thumbnail: string;
        author: Author;
    }

    interface PlaylistItem {
        title: string;
        videoId: string;
        listId: string;
        thumbnail: string;
        author: Author;
    }

    interface PlaylistMetadataResult {
        title: string;
        listId: string;
        url: string;
        views: number;
        date: string;
        image: string;
        thumbnail: string;
        videos: PlaylistItem[];
        author: Author;
    }
}
```
</details>